source/library/Autoexporter.hs

```diff
@@ -16,17 +20,35 @@ main = do
 mainWithArgs :: [String] -> IO ()
 mainWithArgs args =
   case args of
-    [name, inputFile, outputFile] -> autoexport name inputFile outputFile
+    [name, inputFile, outputFile, "--deep"] ->
+      autoexport ExportScope'Deep name inputFile outputFile
+    [name, inputFile, outputFile] ->
+      autoexport ExportScope'Shallow name inputFile outputFile
     _ -> fail ("unexpected arguments: " ++ show args)
 
-autoexport :: String -> FilePath -> FilePath -> IO ()
-autoexport name inputFile outputFile = do
+autoexport :: ExportScope -> String -> FilePath -> FilePath -> IO ()
+autoexport exportScope name inputFile outputFile = do
   let moduleName = makeModuleName name
   let directory = FilePath.dropExtension inputFile
-  files <- Directory.getDirectoryContents directory
+  files <- findFiles exportScope directory
   let output = makeOutput moduleName directory files
   writeFile outputFile output
 
+findFiles :: ExportScope -> FilePath -> IO [FilePath]
+findFiles ExportScope'Shallow dir = filter isHaskellFile <$> findRootFiles dir
+findFiles ExportScope'Deep dir = do
+  rootItems <- findRootFiles dir
+  childItems <- Traversable.for rootItems $ \item -> do
+    let path = dir FilePath.</> item
+    dirExists <- Directory.doesDirectoryExist path
+    if dirExists
+      then fmap (item FilePath.</>) <$> findFiles ExportScope'Deep path
```

Comment:
Same deal here with `<$>`. 

Also: Maybe the "deep" branch of this function should be a separate helper function? That way this recursive call doesn't have to provide the export scope.